homeassistant/components/gree/switch.py

```diff
@@ -75,6 +76,42 @@ async def async_turn_off(self, **kwargs: Any) -> None:
         self.async_write_ha_state()
 
 
+class GreeHealthModeSwitchEntity(GreeEntity, SwitchEntity):
+    """Representation of the health mode on the device."""
+
+    def __init__(self, coordinator):
+        """Initialize the Gree device."""
+        super().__init__(coordinator, "Health mode")
+        self._attr_entity_registry_enabled_default = False
+
+    @property
+    def icon(self) -> str | None:
+        """Return the icon for the device."""
+        return "mdi:pine-tree"
+
+    @property
+    def device_class(self):
+        """Return the class of this device, from component DEVICE_CLASSES."""
+        return SwitchDeviceClass.SWITCH
```

Comment:
My initial recommendation was going to be that these should be static class attributes:
```python
class GreeHealthModeSwitchEntity(GreeEntity, SwitchEntity):
    """Representation of the health mode on the device."""

    _attr_icon = "mdi:pine-tree"
    _attr_device_class = SwitchDeviceClass.SWITCH

    def __init__(self, coordinator):
        ...
```

But now I seee that all other entities in Gree have the same pattern.
I think it should first be migrated to use `SwitchEntityDescription` to remove all the duplicate code.